tutorials/shaders/compute_shaders.rst

```diff
@@ -0,0 +1,242 @@
+.. _doc_compute_shaders:
+
+Using ComputeShaders
+====================
+
+Intro
+-----
+
+Unlike most shaders compute shaders don't have a defined functionality and thus are independent from the graphics pipeline.
+We can think of them as blocks of code executed on the GPU for about any purpose we want.
+The big benefit compared to code executed on a CPU is the high amount of parallelization that GPUs provide.
+
+But because compute shaders are independent of the graphics pipeline we don't have any user defined inputs or outputs. Instead they make changes directly on the GPUs memory which we can read and write  using scripts.
+
+How they work
+-------------
+
+Compute shaders can be thought of as a mass of small computers called work groups.
+Much like super computers they are aligned in rows and columns but also stacked on top of each other
+essentially forming a 3D array of them.
+
+When creating a compute shader we can specify the number of work groups we wish to use.
+Keep in mind that these work groups are independent from each other and therefore can not depend on the result of another one.
+
+In each work group we have another 3D array of threads called invocations, but unlike work groups, invocations can communicate with each other.
+
+So now lets work with a compute shader to see how it really works.
+
+Creating a ComputeShader
+------------------------
+
+To begin using compute shaders, create a new text file called "compute_example.glsl" GLSL is the shader language used in Godot so if you are familiar with normal shaders the syntax below could look somehow familiar.
+
+Let's take a look at this compute shader code:
+
+.. code-block:: glsl
+
+    #[compute]
+    #version 450
+
+    // Invocations in the (x, y, z) dimension 
+    layout(local_size_x = 4, local_size_y = 1, local_size_z = 1) in;
+
+    // A binding to the buffer we create in our script
+    layout(set = 0, binding = 0, std430) restrict buffer MyDataBuffer {
+        double data[];
+    }
+    my_data_buffer;
+
+    // The code we want to execute in each invocation
+    void main() {
+        // gl_GlobalInvocationID.x uniquely identifies this invocation within the work group
```

Comment:
This isn't quite accurate, gl_GlobalInvocationID.x uniquely identifies this invocation over all work groups, not within a given work group